### DISCUSSION TITLE
Eigenstrain base

### DISCUSSION THREAD

**Discussion by RossMech** [idaholab/moose#16088] [Q&A Modules: General] [answered]
Hello everyone,

I have a question about the eigenstrain_base:
https://mooseframework.inl.gov/source/materials/ComputeEigenstrain.html

What is an order in eigenstrain_base vector to a second order tensor? I have a symmetric tensor, which includes tension-compression and shear strains in 2D and 3D. It would be also very handily to add this information to a documentation of software.

Best regards,
Rostyslav Nizinkovskyi.

**Comment by dschwen** [comment, marked as answer]
If you specify 6 components the order is the [Voigt notation](https://en.wikipedia.org/wiki/Voigt_notation). Other fill patterns can be seen [here in the source code](https://github.com/idaholab/moose/blob/next/framework/include/utils/RankTwoTensorImplementation.h#L146). I thought we had this in the docs, but I think I was remembering the RankFourTensor instead. We should add this.

**Reply by RossMech** [reply]
Thank you for the answer!
